2.adoc

```diff
@@ -72,5 +72,229 @@ Optional description
+
+link:https://ace.c9.io/[]
+
+==== Porovnání rozdílů
+
+. Ace je čistě editor pro text/kód
+. Nejsou zde nijak řešeny markup-specifické záležitosti jako outline, preview, atp.
+. Jedná se o velmi propracovaný textový editor s velkou rozšířitelností
+
+==== Zhodnocení UI
+
+Pro lepší porovnání na pár místech ukazuji i praktické nasazení Ace editoru v různých nástrojích.
+
+===== Základní pohled
+
+Webová stránka Ace nabízí takzvanou "kuchyň" na které si uživatel může vyzkoušet různé základní vlastnosti:
+
+image:/assets/ace-zakladni-ui.png[]
+
+Editor v základní podobě nabízí spoustu vlastností:
+
+. syntax highlighting
+. číslování řádků
+. možnost zabalovat/rozbalovat text podle syntaxe
+. zobrazování neviditelných znaků
+. zobrazování aktuální pozice (pravý dolní roh), případně i pozici označeného textu
+. soft wrap indikátor (šedá vertikální čára v 80 sloupci textu)
+
+Rovněž umožňuje i barvení code snippetů uvnitř markup code bloků:
+
+image:/assets/ace-barveni-v-kodu.png[]
+
+===== Klávesové ovládání
+
+Editor sám o sobě nemá žádné klávesové zkratky pro vkládání syntaktických konstruktů, nicméně má velmi propracovanou klávesovou podporu pro editaci textu. Lze například nastavit mód _vim_, v kterém lze používat vim přikazy nejen pro navigaci (`3w`, `10j`, `67G`, …), ale i pro editaci textu (`dap`, `di{)`, `3yy`, `p`, …), což obzvlášť ocení náročnější uživatelé.
+Dokonce podporuje i některé ex příkazy jako například search & replace:
+
+image:/assets/ace-vim-ex.png[]
+
+Ace umožňuje plně konfigurovat klásové zkratky pro běžný mód, i pro speciální módy jako je již zmíněný vim, či emacs.
+
+===== Náhled změn
+
+Ace rovněž nabízí možnost zobrazování rozdílů v textu/mezi verzemi. Zde například editor použitý na webu GitHub v read-only módu zobrazující změny mězi dvěma verzemi souboru.
+
+image:/assets/ace-nahled-zmen.png[]
+
+===== Rozšiřování editoru
+
+Jednou ze silných stránek editoru je rozšířitelnost a tudíž i široká aplikovatelnost.
+
+image:/assets/ace-rstudio.png[]
+
+Na obrázku je Ace editor embedovaný do nástroje R Studio. Uvnitř editoru je každý `R` code blok obohacen o tlačítka ke konfiguraci či spuštění. Po spuštění uvedeného příkladu se přímo do editoru vloží vykreslený graf.
+
+==== Pozorování UI v kontextu markup-editor
+
+Ace editor lze srovnat pouze s textovou částí markup-editoru. V této části se významně překrývají.
+
+. vyhledávání v dokumentu
+. klávesové zkratky
+. syntax highlighting
+. status bar
+. diff view
+. číslování řádek
+. a mnoho dalšího
+
+Ace by bylo možno zvolit i jako základ pro textový editor markup-editoru.
+
+==== Shrnutí
+
+===== Plusy
+. Bohatá rozšířitelnost se spoustou pluginů
+. I v základní podobě velmi silný textový editor
+. Možnost zobrazovat rich content (např. obrázky) přímo v editoru
+. vim mód :)
+
+===== Mínusy
+. Nutná značná investice do učení se API pro tvorbu vlastních rozšíření
+
```

Comment:
Chapu, ze absence outline, preview, atd. neni minus, kdyz se na to editor nezameruje. Ale presto bych to uvedl jako minus v kontextu semestralky.

Reply:
přidáno